stock_inventory_valuation_report/reports/stock_inventory_valuation_report.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockInventoryValuationView(models.TransientModel):
+    _name = "stock.inventory.valuation.view"
+    _description = "Stock Inventory Valuation View"
+
+    name = fields.Char()
+    reference = fields.Char()
+    barcode = fields.Char()
+    qty_at_date = fields.Float()
+    uom_id = fields.Many2one(
+        comodel_name="uom.uom",
+    )
+    currency_id = fields.Many2one(
+        comodel_name="res.currency",
+    )
+    cost_currency_id = fields.Many2one(
+        comodel_name="res.currency",
+    )
+    standard_price = fields.Float()
+    stock_value = fields.Float()
+    cost_method = fields.Char()
+
+
+class StockInventoryValuationReport(models.TransientModel):
+    _name = "report.stock.inventory.valuation.report"
+    _description = "Stock Inventory Valuation Report"
+
+    # Filters fields, used for data computation
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+    )
+    compute_at_date = fields.Integer()
+    date = fields.Datetime()
+
+    # Data fields, used to browse report data
+    results = fields.Many2many(
+        comodel_name="stock.inventory.valuation.view",
+        compute="_compute_results",
+        help="Use compute fields, so there is nothing store in database",
+    )
+
+    def _compute_results(self):
+        self.ensure_one()
+        if not self.compute_at_date:
+            self.date = fields.Datetime.now()
+        products = (
+            self.env["product.product"]
+            .with_context(
+                to_date=self.date, company_owned=True, create=False, edit=False
+            )
+            .search([("type", "=", "product"), ("qty_available", "!=", 0)])
+        )
+        ReportLine = self.env["stock.inventory.valuation.view"]
+        for product in products:
+            standard_price = product.standard_price
+            line = {
+                "name": product.name,
+                "reference": product.default_code,
+                "barcode": product.barcode,
+                "qty_at_date": product.qty_available,
```

Comment:
should also likely be `quantity_svl`

Reply:
Thank you for your feedback
You are right
I fix it, please check again